src/classes/AggregateResultQueryBuilder.cls

```diff
@@ -0,0 +1,188 @@
+        return this.max(queryField, null);
+    }
+
+    public IAggregateResultQueryBuilder max(IQueryField queryField, String fieldAlias) {
+        return buildAggregateFunction('MAX', queryField, fieldAlias);
+    }
+
+    /**
+    * @description Adds the minimum value of the field to the dynamically generated aggregate query
+    * @param queryField The field to use for calculating the minimum
+    * @return The instance of IAggregateResultQueryBuilder, to allow chaining methods
+    */
+    public IAggregateResultQueryBuilder min(IQueryField queryField) {
+        return this.min(queryField, null);
+    }
+
+    public IAggregateResultQueryBuilder min(IQueryField queryField, String fieldAlias) {
+        return buildAggregateFunction('MIN', queryField, fieldAlias);
+    }
+
+    /**
+    * @description Sums the values of the supplied numeric field to the dynamically generated aggregate query
+    * @param numericQueryField The field to use for calculating the minimum
+    * @return The instance of IAggregateResultQueryBuilder, to allow chaining methods
+    */
+    public IAggregateResultQueryBuilder sum(IQueryField numericQueryField) {
+        return this.sum(numericQueryField, null);
+    }
+
+    public IAggregateResultQueryBuilder sum(IQueryField numericQueryField, String fieldAlias) {
+        return buildAggregateFunction('SUM', numericQueryField, fieldAlias);
+    }
+
+    public IAggregateResultQueryBuilder filterBy(IQueryFilter queryFilter) {
+        super.doFilterBy(queryFilter);
+        return this;
+    }
+
+    public IAggregateResultQueryBuilder filterBy(List<IQueryFilter> queryFilters) {
+        super.doFilterBy(queryFilters);
+        return this;
+    }
+
+    public IAggregateResultQueryBuilder orderBy(IQueryField orderByQueryField) {
+        super.doOrderBy(orderByQueryField);
+        return this;
+    }
+
+    public IAggregateResultQueryBuilder orderBy(IQueryField orderByQueryField, QuerySortOrder sortOrder) {
+        super.doOrderBy(orderByQueryField, sortOrder);
+        return this;
+    }
+
+    public IAggregateResultQueryBuilder orderBy(IQueryField orderByQueryField, QuerySortOrder sortOrder, QueryNullSortOrder nullsSortOrder) {
+        super.doOrderBy(orderByQueryField, sortOrder, nullsSortOrder);
+        return this;
+    }
+
+    public IAggregateResultQueryBuilder limitCount(Integer limitCount) {
+        super.doLimitCount(limitCount);
+        return this;
+    }
+
+    public String getQuery() {
+        String queryString = 'SELECT ' + this.getGroupByFieldString(false) + this.getAggregateFunctionString()
+            + '\nFROM ' + this.sobjectType.getDescribe().getName()
+            + super.doGetWhereClauseString()
+            + this.getGroupByFieldString(true)
+            + super.doGetOrderByString()
+            + super.doGetLimitCountString();
+
+        return queryString;
+    }
+
+    public AggregateResult getFirstQueryResult() {
```

Comment:
did you try taking these out?  https://developer.salesforce.com/docs/atlas.en-us.api.meta/api/sforce_api_calls_query_aggregateresult.htm - aggregate result is just a read-only SObject

Reply:
Yo! Thanks for reviewing! I think I'm going to keep these methoda for now, just for the sake of clarity and being explicit when possible